tools/typelib-clang-tlb-importer/CMakeLists.txt

```diff
@@ -1,116 +1,92 @@
+include(LLVMTools)
 
-# prevent loud warnings from gcc-4.9 caused by a macro-expansion the
-# AST-headers... this only clutters up the compiler output with nonsense
-add_definitions("-Wno-unused-local-typedefs")
+# we tested and developed _only_ against llvm-3.4
+llvm_find_config(3.4)
 
-# when compiling inside the rock-framework "Wall" is used. This adds alotta
-# more bogus warnings from Debians llvm-3.4 headers...
-add_definitions("-Wno-strict-aliasing")
+if (NOT LLVM_CONFIG_EXECUTABLE)
+    message(STATUS "typelib-clang-tlb-importer: llvm-config was not found, skipping clang-based importer")
+else()
+    message(STATUS "typelib-clang-tlb-importer: configuring with ${LLVM_CONFIG_EXECUTABLE}")
+
+    # note that "cxxflags" additionally returns the "-I" flags... which might not
+    # be what you want!
+    llvm_get_config(LLVM_CXXFLAGS "--cxxflags")
+    string(REGEX REPLACE "-I[^ ]+" "" LLVM_CXXFLAGS ${LLVM_CXXFLAGS})
+
+    llvm_get_config(LLVM_LIBRARY_DIRS "--libdir")
+
+    # tell cmake where to look for includes
+    llvm_get_config(LLVM_INCLUDE_DIRS "--includedir")
+
+    llvm_get_config(llvm_config_libs "--libs"
+        core option analysis support bitreader mcparser transformutils)
+    # Convert the library as reported by llvm-config into a cmake list
+    string(REPLACE "-l" "" llvm_config_libs "${llvm_config_libs}")
+    string(REPLACE " " ";" llvm_config_libs "${llvm_config_libs}")
+    # and add it to LLVM_RAW_LIBRARIES
+    list(APPEND LLVM_RAW_LIBRARIES ${llvm_config_libs})
+
+    foreach(llvm_lib ${LLVM_RAW_LIBRARIES})
+        unset(llvm_lib_path CACHE)
+        find_library(llvm_lib_path ${llvm_lib} PATHS ${LLVM_LIBRARY_DIRS})
+        if (NOT llvm_lib_path)
+            message(STATUS "${llvm_lib} not found in ${LLVM_LIBRARY_DIRS}")
+            set(LLVM_LIBRARIES FALSE)
+            break()
+        else()
+            list(APPEND LLVM_LIBRARIES ${llvm_lib_path})
+        endif()
+    endforeach()
 
-macro(llvm_find_config LLVM_REQUIRED_VERSION)
-    if (LLVM_REQUIRED_VERSION STREQUAL "")
-        list(APPEND __llvm_names llvm-config-3.6 llvm-config-3.5 llvm-config-3.4)
-    else()
-        list(APPEND __llvm_names llvm-config-${LLVM_REQUIRED_VERSION})
-    endif()
-    message("-- llvm: looking for ${__llvm_names}")
-    find_program(LLVM_CONFIG_EXECUTABLE NAMES ${__llvm_names})
-    if (NOT LLVM_CONFIG_EXECUTABLE)
-        message(FATAL_ERROR "could not find the llvm-config script (candidates: ${__llvm_names})")
-    endif()
-    message("-- llvm: found ${LLVM_CONFIG_EXECUTABLE}")
-endmacro()
-
-macro(llvm_get_config OUTPUT)
-    execute_process(COMMAND ${LLVM_CONFIG_EXECUTABLE} ${ARGN}
-        RESULT_VARIABLE __llvm_ok
-        OUTPUT_VARIABLE ${OUTPUT})
-    if (NOT __llvm_ok STREQUAL 0)
-        message(FATAL_ERROR "failed to execute llvm-config ${ARGN} (result: ${__llvm_ok})")
+    if (NOT LLVM_LIBRARIES)
+        message(WARNING "typelib-clang-tlb-importer: some libraries cannot be found...?")
```

Comment:
You're still building even though you know some libraries are missing. Was that on purpose ?